packages/type-graphql/tests/functional/middlewares.ts

```diff
@@ -60,7 +60,8 @@ describe("Middlewares", () => {
     };
     const errorCatchMiddleware: MiddlewareFn = async (_, next) => {
       try {
-        return await next();
+        const result = await next();
+        return result;
```

Comment:
`return await` is a valid statement. I can't agree of linter forcing own version for no reason. return without await in try-catch is an error

Reply:
[The ESLint rule](https://eslint.org/docs/latest/rules/no-return-await) explains the pros and cons of `return await`

Reply:
It's literally the case in `tests/functional/middlewares.ts`:
```
// In this example the `await` is necessary to be able to catch errors thrown from `bar()`
async function foo() {
    try {
        return await bar();
    } catch (error) {}
}
```

Reply:
So it's something wrong with the eslint setup that it throws error about `await` for non-promise? Because that was the error I was talking with @carlocorradini about 
![Uploading image.png…]()

Reply:
I'm thinking this happened:
```ts
// ESLint error, no-return-await
return await next()

// ESLint wants
return next()

// This also removes the error, and keeps the stack trace at the cost of an extra microtask
const result = await next()
return result
```